From 5416d85fdc552639144a76d0f31ea6610b76994f Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Wed, 8 Jul 2026 16:58:13 +0200
Subject: [PATCH 32/46] AnnaThermostat: add failed_burner_flame_ignitions
 measurement

Map failed_burner_flame_ignitions to a Domoticz custom counter-style numeric sensor.

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>
Assisted-by: GPT-5.3-Codex

Upstream-Status: Inappropriate [WIP]
---
 hardware/AnnaThermostat.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/hardware/AnnaThermostat.cpp b/hardware/AnnaThermostat.cpp
index 7f7649b72..9b1f0cdc4 100644
--- a/hardware/AnnaThermostat.cpp
+++ b/hardware/AnnaThermostat.cpp
@@ -1303,6 +1303,14 @@ void CAnnaThermostat::GetMeterDetails()
 					SendCustomSensor(applianceIDhex, 1, batterypercentage, starts, sname, "");
                 }
             }
+			else if (sname == "failed_burner_flame_ignitions")
+			{
+				if (!tmpstr.empty())
+				{
+					float ignitions = (float)atof(tmpstr.c_str());
+					SendCustomSensor(applianceIDhex, 14, batterypercentage, ignitions, sname, "");
+				}
+			}
             else if (sname == "burner_operation_time")
             {
                 if (!tmpstr.empty())
